File: FLASH-FLOOD/backend/app/schemas/weather.py
from pydantic import BaseModel
from .common import DemoMetadata


class WeatherCurrentResponse(DemoMetadata):
    status: str
    source: str | None
    updated_at: str | None
    location_id: str
    location_name: str
    latitude: float | None
    longitude: float | None
    temperature: float | None
    humidity: float | None
    current_rainfall: float | None
    rainfall_1h: float | None
    rainfall_3h: float | None
    rainfall_6h: float | None
    rainfall_24h: float | None
    rainfall_72h: float | None
    rainfall_7d: float | None
    antecedent_rainfall_7d: float | None
    precipitation_probability: float | None
    forecast_rainfall: float | None
    wind_speed: float | None
    wind_direction: str | None
    pressure: float | None
    soil_moisture: float | None
    soil_moisture_0_to_7cm: float | None
    hourly_rainfall: list[float] | None
    scenario: str | None
    note: str | None


class WeatherForecastResponse(DemoMetadata):
    status: str
    source: str | None
    updated_at: str | None
    location_id: str
    location_name: str
    latitude: float | None
    longitude: float | None
    forecast_rainfall: float | None
    precipitation_probability: float | None
    hourly_rainfall: list[float] | None
    entries: list[dict[str, str | float | None]] | None
    scenario: str | None
    note: str | None

File: frontend/src/types/weather.ts
export interface WeatherSnapshot { temperature: number; humidity: number; currentRainfall: number; rainfall1h: number; rainfall3h: number; rainfall6h: number; rainfall24h: number; forecastRainfall: number; precipitationProbability: number; windSpeed: number; windDirection: string; pressure: number; hourlyRainfall: number[] }

/**
 * Risk-assessment weather as consumed by the command UI. Values are nullable
 * because live provider values (and the ML windows) are honestly unavailable
 * until measured — a missing value is shown as "Unavailable", never as 0.
 */
export interface CommandWeatherValue {
  temperature: number | null
  humidity: number | null
  currentRainfall: number | null
  rainfall1h: number | null
  rainfall3h: number | null
  rainfall6h: number | null
  rainfall24h: number | null
  /** ML/augmented windows — present on live ML risk responses, absent from demo fixtures. */
  rainfall72h?: number | null
  rainfall7d?: number | null
  antecedentRainfall7d?: number | null
  forecastRainfall: number | null
  precipitationProbability: number | null
  windSpeed: number | null
  windDirection: string | null
  pressure: number | null
  soilMoisture?: number | null
  soilMoisture0to7cm?: number | null
  hourlyRainfall: number[] | null
}
